### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by jorgegoco**
- Added unit tests for all the methods in all the classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - all classes created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    it 'test can_use_services method' do
+      @person.age = 17
+      @person.parent_permission = false
+      expect(@person.can_use_services?).to be false
+    end
+
+    it 'test of_age method' do
+      expect(@person.of_age?).to be true
+    end
+  end
```
Comment (Whoistolu) [inline spec/person_spec.rb]: - [ ] [REQUIRED] Great job team, as stated by the project requirement, kindly add tests for all methods in this class. `correct_name` and the `add_rental` methods.

File: spec/student_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      expect(@student.age).to be < 18
+      expect(@student.name).to eq 'test_name'
+      expect(@student.parent_permission).to be false
+      expect(@student.id).to be_between(1, 1000)
+    end
+
+    it 'test play_hooky method' do
+      expect(@student.play_hooky).to eq '¯\(ツ)/¯'
+    end
+  end
```
Comment (Whoistolu) [inline spec/student_spec.rb]: - [ ] [REQUIRED] Kindly write a test for the `classroom` method as well.

File: spec/book_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+describe Book do
+  context 'Unit testing in the Book class' do
+    it 'test creating an instance' do
+      book = Book.new('test_title', 'test_author')
+      expect(book.title).to eq 'test_title'
+      expect(book.author).to eq 'test_author'
+      expect(book.rentals).to eq []
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/book_spec.rb]: - [ ] [REQUIRED] Great job so far team 👍🏽 . Kindly add a test for the `add_rental` method as stated by the project requirements.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi Team,
@jorgegoco and @maaloum 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._
- [ ] [REQUIRED] I am putting this recommendations here because we don not ave them in the file changes. Let's add tests for the following classes as stated by the project requirements : `decorator.rb`, `nameable.rb`, `capitalize_decorator`, and the `trimmer_decorator`.

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @jorgegoco & @maaloum , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- Professional Readme File ✔✔
- Linters are setup correctly ✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**